pkg/utils/utils.go

```diff
@@ -44,3 +45,16 @@ func ContextWithDeadlineFn(ctx context.Context, deadlineFn func(orig time.Time)
 	}
 	return ctx, cancel
 }
+
+func IsZero[C comparable](val C) bool {
+	var zero C
+	return zero == val
+}
+
+// AddHexPrefix adds the prefix (0x) to a given hex string.
+func AddHexPrefix(str string) string {
+	if len(str) < 2 || len(str) > 1 && strings.ToLower(str[0:2]) != "0x" {
```

Comment:
How about std lib for readability? (should still be plenty fast too)
```suggestion
	if !strings.HasPrefix(s, "0x") {
```